projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php

```diff
@@ -499,6 +499,22 @@ public function get_settings_response() {
 
 	}
 
+	/**
+	 * Get the default value for the wpcom_gifting_subscription option.
+	 * The default value is the inverse of the plan's auto_renew setting.
+	 *
+	 * @return bool
+	 */
+	protected function get_wpcom_gifting_subscription_default() {
+		if ( function_exists( 'wpcom_site_has_feature' ) ) {
```

Comment:
Does checking if `Gifting_Banner` exists makes more sense or do we need to check for both maybe?

Reply:
Oh yeah, I think you be right. Check for Gifting_Banner is probably better.

Reply:
I think it makes sense to check the plan has the feature as well (doesn't make sense to return true if it doesn't) but then again I don't know if it matters. Will anything consume a true value for plans that don't have the feature?

Reply:
Reworked this a bit to use `wpcom_get_site_purchases`, `wpcom_purchase_has_feature`, and check for the `subscription-gifting` feature, because that seems more reliable than using `Gifting_Banner::get_plan_purchase` (which is also yet to be merged in wpcomsh).